giskard_vision/landmark_detection/detectors/head_pose_detector.py

```diff
@@ -0,0 +1,35 @@
+from giskard_vision.landmark_detection.dataloaders.wrappers import (
+    CachedDataLoader,
+    FilteredDataLoader,
+    HeadPoseDataLoader,
+)
+
+from .base import LandmarkDetectionBaseDetector
+from .decorator import maybe_detector
+
+
+@maybe_detector("headpose_landmark", tags=["landmark"])
+class HeadPoseDetectorLandmark(LandmarkDetectionBaseDetector):
+    """
+    Detector that evaluates models performance depending on the head position
+    """
+
+    group: str = "Head Pose"
+
+    def get_dataloaders(self, dataset):
+        cached_dl = CachedDataLoader(HeadPoseDataLoader(dataset), cache_size=None, cache_img=False, cache_marks=False)
+
+        head_poses = [("positive roll", self._positive_roll), ("negative roll", self._negative_roll)]
+        dls = []
+
+        for hp in head_poses:
+            current_dl = FilteredDataLoader(cached_dl, hp[1])
+            dls.append(current_dl)
+
+        return dls
+
+    def _positive_roll(self, elt):
+        return elt[2]["headPose"]["roll"] > 0
+
+    def _negative_roll(self, elt):
+        return elt[2]["headPose"]["roll"] < 0
```

Comment:
can we extend these a bit?
1. roll == 0

and:
1.  -90 <= yaw < -45
2. -45 <= yaw < 0
3. yaw == 0
4. 0 < yaw <= 45
5. 45 < yaw <= 90

and also:
1. pitch > 0
2. pitch < 0
3. pitch == 0